src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs

```diff
@@ -192,19 +192,7 @@ public AutoCompleteSource AutoCompleteSource
             }
             set
             {
-                if (!ClientUtils.IsEnumValid_NotSequential(value, (int)value,
-                                                    (int)AutoCompleteSource.None,
-                                                    (int)AutoCompleteSource.AllSystemSources,
-                                                    (int)AutoCompleteSource.AllUrl,
-                                                    (int)AutoCompleteSource.CustomSource,
-                                                    (int)AutoCompleteSource.FileSystem,
-                                                    (int)AutoCompleteSource.FileSystemDirectories,
-                                                    (int)AutoCompleteSource.HistoryList,
-                                                    (int)AutoCompleteSource.ListItems,
-                                                    (int)AutoCompleteSource.RecentlyUsedList))
-                {
-                    throw new InvalidEnumArgumentException(nameof(value), (int)value, typeof(AutoCompleteSource));
-                }
+                EnumValidation.EnumValidator.Validate(value);
```

Comment:
Do you think we could have a different name that better communicates the validation?
E.g. something like:
```suggestion
                EnumValidation.EnumValidator.ValidateNotSequential(value);
```

Reply:
It's your codebase, but personally I think the power of the source generator is that it doesn't need this sort of distinction in order to work. You could create such a distinction, but you'd just end up with compile errors that amounted to: "You've done A. You meant to do B. I know you meant to do B, its the only option here, but I'm going to fail until you change the code to say 'B'". I think it would be annoying.

Reply:
@JeremyKuhne what do you think?

Reply:
> It's your codebase, but personally I think the power of the source generator is that it doesn't need this sort of distinction in order to work.

I agree with this point

Reply:
> what do you think?

I don't think we should have different names. Above and beyond @davidwengier's point, perf notwithstanding we probably wouldn't create a generic `ValidateFlagsEnum<T>()` and `ValidateNonFlagsEnum<T>()`, nor would we want to go into significant detail `ValidateEnumValueIsDefined` or `ValidateFlagEnumValuesAreAllDefined`. It isn't terrible, but it doesn't add much value to the comprehension of the calling method and, in fact, may actually be more distracting than helpful.